metricshub-engine/src/main/java/org/sentrysoftware/metricshub/engine/strategy/AbstractStrategy.java

```diff
@@ -421,11 +422,48 @@ protected boolean validateConnectorDetectionCriteria(final Connector currentConn
 			String.format(CONNECTOR_ID_FORMAT, KnownMonitorType.CONNECTOR.getKey(), connectorId)
 		);
 
-		collectConnectorStatus(connectorTestResult.isSuccess(), connectorId, monitor);
+		// Add statusInformation to legacyTextParameters attribute of the connector monitor
+		final String statusInformation = buildStatusInformation(hostname, connectorTestResult);
+		final Map<String, String> legacyTextParameters = monitor.getLegacyTextParameters();
+		legacyTextParameters.put("StatusInformation", statusInformation);
 
+		collectConnectorStatus(connectorTestResult.isSuccess(), connectorId, monitor);
 		return connectorTestResult.isSuccess();
 	}
 
+	/**
+	 * Builds the status information for the connector
+	 * @param hostname   Hostname of the resource being monitored
+	 * @param testResult Test result of the connector
+	 * @return String representing the status information
+	 */
+	protected String buildStatusInformation(final String hostname, final ConnectorTestResult testResult) {
+		final StringBuilder value = new StringBuilder();
+
+		final String builtTestResult = testResult
+			.getCriterionTestResults()
+			.stream()
+			.map(criterionResult -> {
+				final String result = criterionResult.getResult();
+				final String message = criterionResult.getMessage();
+				return String.format(
```

Comment:
Each extension provides the messages in a specific format. We need to normalize all the messages. It is a huge work but we should do that to get normalized messages for BMC or even for OTEL logs.